### ISSUE TITLE
Dynamic importing fails inside PEX file

### ISSUE THREAD

**Issue by DrumMachineLearning** [pantsbuild/pex#446]
TL;DR: I cannot dynamically import submodules inside a PEX file because my discovery function is looking at the local filesystem instead of in the folders inside the PEX.

Long version: I am managing a project that consists of a central CLI framework we would like to distribute as a PEX. The CLI's commands are all submodules that are managed independently of the central project.

As I have no control over the commands, they are loaded dynamically as follows:

```python
def load_commands(plugin_folder, group):
    for folder in [x for x in os.listdir(plugin_folder) if os.path.isdir(os.path.join(plugin_folder, x))]:
        for fname in [x for x in os.listdir(os.path.join(plugin_folder, folder)) if x.endswith('_main.py')]:
            name = fname[:-8]
            try:
                module = __import__('.'.join(('my_project', 'commands', name, fname[:-3])), None, None, [name])
                cmd = getattr(module, name)
                group.add_command(cmd)
            except ImportError as e:
                announce_import_problem(e)
```

This works fine in an unpackaged distribution. But as soon as we try to run from inside a PEX, we get this:

`OSError: [Errno 20] Not a directory: '/Users/MyUsername/dist/my_pex.pex/my_project/commands'`

**Comment by jsirois** [comment]
Have you tried creating the pex with ` --not-zip-safe`? This, or running with `PEX_FORCE_LOCAL=1 ./<pex>` will force the pex to extract itself to the filesystem before running.

**Comment by jsirois** [comment]
I'm going to assume not and mark this as answered. Please re-open with more details or questions if `--not-zip-safe` is not an acceptable answer for you here.

**Comment by jsirois** [comment]
Yup, the `python_binary` target has `zip_safe` `bool` parameter - just pass `False`. See the BUILD dictionary for more of this sort of thing: https://www.pantsbuild.org/build_dictionary.html
